bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -0,0 +1,234 @@
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/annotations"
+
+	bootstrapv1 "sigs.k8s.io/cluster-api-provider-aws/bootstrap/eks/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-aws/bootstrap/eks/internal/userdata"
+)
+
+// EKSConfigReconciler reconciles a EKSConfig object
+type EKSConfigReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+type EKSConfigScope struct {
+	logr.Logger
+	Config      *bootstrapv1.EKSConfig
+	ConfigOwner *bsutil.ConfigOwner
+	Cluster     *clusterv1.Cluster
+}
+
+// +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=eksconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=eksconfigs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusters;awsmanagedclusters,verbs=get;list;watch;
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=machines;machinepools;clusters,verbs=get;list;watch;
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;delete;
+
+func (r *EKSConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("eksconfig", req.NamespacedName)
+
+	// get EKSConfig
+	config := &bootstrapv1.EKSConfig{}
+	if err := r.Client.Get(ctx, req.NamespacedName, config); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get config")
+		return ctrl.Result{}, err
+	}
+
+	// check owner references and look up owning Machine object
+	configOwner, err := bsutil.GetConfigOwner(ctx, r.Client, config)
+	if apierrors.IsNotFound(err) {
+		// no error here, requeue until we find an owner
+		return ctrl.Result{}, nil
+	}
+	if err != nil {
+		log.Error(err, "Failed to get owner")
+		return ctrl.Result{}, err
+	}
+	if configOwner == nil {
+		// no error, requeue until we find an owner
+		return ctrl.Result{}, nil
+	}
+
+	log = log.WithValues("kind", configOwner.GetKind(), "version", configOwner.GetResourceVersion(), "name", configOwner.GetName())
+
+	cluster, err := util.GetClusterByName(ctx, r.Client, configOwner.GetNamespace(), configOwner.ClusterName())
+	if err != nil {
+		if errors.Cause(err) == util.ErrNoCluster {
+			log.Info(fmt.Sprintf("%s does not belong to a cluster yet, requeueing until it's part of a cluster", configOwner.GetKind()))
+			return ctrl.Result{}, nil
+		}
+		if apierrors.IsNotFound(err) {
+			log.Info("Cluster does not exist yet, requeueing until it is created")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Could not get cluster with metadata")
+		return ctrl.Result{}, err
+	}
+
+	if annotations.IsPaused(cluster, config) {
+		log.Info("Reconciliation is paused for this object")
+		return ctrl.Result{}, nil
+	}
+
+	scope := &EKSConfigScope{
+		Logger:      log,
+		Config:      config,
+		ConfigOwner: configOwner,
+		Cluster:     cluster,
+	}
```

Comment:
So, getting rid of the EKSConfigScope type entirely? I think I agree. We initially copied this to follow the pattern set forth in the Kubeadm provider but the logic in this controller is simpler so it may be a bit much.

Reply:
Think I'm in favour of cleaning this up as there's inconsistent use of the logger from either the reconciler or the scope (see below).

Reply:
sounds good, will do